File: pkg/codegen/testing/test/testdata/docker-func-invoke-pp/nodejs/docker-func-invoke.ts
import * as pulumi from "@pulumi/pulumi";
import * as docker from "@pulumi/docker";

const latest = docker.getRemoteImage({
    name: "nginx",
});
const ubuntu = new docker.RemoteImage("ubuntu", {name: "ubuntu:precise"});
export const remoteImageId = latest.then(latest => latest.id);
export const ubuntuImage = ubuntu.name;

File: pkg/codegen/testing/test/testdata/docker-func-invoke-pp/dotnet/docker-func-invoke.cs
using System.Collections.Generic;
using Pulumi;
using Docker = Pulumi.Docker;

return await Deployment.RunAsync(() => 
{
    var latest = Docker.GetRemoteImage.Invoke(new()
    {
        Name = "nginx",
    });

    var ubuntu = new Docker.RemoteImage("ubuntu", new()
    {
        Name = "ubuntu:precise",
    });

    return new Dictionary<string, object?>
    {
        ["remoteImageId"] = latest.Apply(getRemoteImageResult => getRemoteImageResult.Id),
        ["ubuntuImage"] = ubuntu.Name,
    };
});

